Adds date math documentation (#2171)
opensearch-project/documentation-website: _opensearch/supported-field-types/date.md

  }
}
```

## Date math

> The date field type supports date math to specify duration in queries. For example, the `gt`, `gte`, `lt`, and `lte` parameters in [range queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/term/#range-query) and the `from` and `to` parameters in [date range aggregations]({{site.url}}{{site.baseurl}}/opensearch/bucket-agg/#range-date_range-ip_range) accept date math expressions.

Reviewer: "supports using date math"?

> A date math expression contains a fixed date, optionally followed by one or more mathematical expressions. The fixed date may be either `now` (current date/time in milliseconds since the epoch) or a string ending with `||` that specifies a date (for example, `2022-05-18||`). The date must be in the `strict_date_optional_time||epoch_millis` format.

Reviewer 2: if it's "date and time" then remove the slash that replaces the word "or" in "date/time"

> Date math supports the following mathematical operators:

Reviewer 3: ```suggestion
Date math supports the following mathematical operators.
```

Operator | Description | Example
:--- | :--- | :---

> `+` | Addition | `+1M`: Add one month

Reviewer: ```suggestion
`+` | Addition | `+1M`: Add 1 month.
```

> `-` | Subtraction | `-1y`: Subtract one year

Reviewer: ```suggestion
`-` | Subtraction | `-1y`: Subtract 1 year.
```

> `/` | Rounding down | `/h`: Round to the beginning of the hour

Reviewer: ```suggestion
`/` | Rounding down | `/h`: Round to the beginning of the hour.
```

> Date math supports the following time units:

Reviewer 3: Is this supposed to be formatted as a table, or a list? If so, replace the colon with a period ("units.")

Reviewer 3: Same for subsequent sections where this may apply.

Author: Not a table; kind of a list. This matches the previous section's unit listing.

`y`: Years<br>
`M`: Months<br>
`w`: Weeks<br>
`d`: Days<br>
`h` or `H`: Hours<br>
`m`: Minutes<br>
`s`: Seconds
{: .note }

### Example expressions

The following example expressions illustrate using date math:

> - `now+1M`: The current date/time in milliseconds since the epoch, plus one month.

Reviewer 2: change to "date and time" unless it's "date or time" (slash replaces "or")

- `2022-05-18||/M`: 05/18/2022, rounded to the beginning of the month. Resolves to `2022-05-01`.
- `2022-05-18T15:23||/h`: 15:23 on 05/18/2022, rounded to the beginning of the hour. Resolves to `2022-05-18T15`.

> - `2022-05-18T15:23:17.789||+2M-1d/d`: 15:23:17.789 on 05/18/2022 plus two months minus one day, rounded to the beginning of the day. Resolves to `2022-07-17`.

Reviewer: ```suggestion
- `2022-05-18T15:23:17.789||+2M-1d/d`: 15:23:17.789 on 05/18/2022, plus 2 months minus 1 day, rounded to the beginning of the day. Resolves to `2022-07-17`.
```

### Using date math in a range query

> The following example shows using date math in a [range query]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/term/#range-query).

Reviewer 3: ```suggestion
The following example shows the use of date math in a [range query]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/term/#range-query).
```

Set up an index with `release_date` mapped as `date`:

```json
PUT testindex 
{
  "mappings" : {
    "properties" :  {
      "release_date" : {
        "type" : "date"
      }
    }
  }
}
```

Index two documents into the index:

```json
PUT testindex/_doc/1
{
  "release_date": "2022-09-14"
}

PUT testindex/_doc/2
{
  "release_date": "2022-11-15"
}
```

> The following query searches for documents with `release_date` within two months and one day of 09/14/2022. The lower boundary of the range is rounded to the beginning of the day on 09/14/2022:

Reviewer: ```suggestion
The following query searches for documents with `release_date` within 2 months and 1 day of 09/14/2022. The lower boundary of the range is rounded to the beginning of the day on 09/14/2022:
```
